### PR TITLE
Add a list of accepted "misspelled" words

### PR DESCRIPTION
**Pull request by IndrajeetPatil**

### CODE REVIEW

**Comment by MichaelChirico** [comment]
at a glance this seems like more of a maintenance burden than it's worth.

Why have this in CI, as opposed to part of the release procedure?

**Comment by IndrajeetPatil** [comment]
I have removed the spelling test to not gain an additional soft dependency.

The idea here is to keep a list of words which we know to be acceptable words, so that we don't need to go through the list every single time we make a release.

Currently, on main, devtools::spell_check() generates a huge list we need to go through to see which are truly misspelled words.

If we add a list like the one in this PR, only the newly introduced, misspelled words will be flagged. This reduces maintainance burden for us.

**Comment by MichaelChirico** [comment]
> we don't need to go through the list every single time we make a release.

right but there will presumably be some churn, every time new documentation/vignettes are added there's likely to be some new words to add.

happy to try it out for a few releases to see just how big that delta is each time though

File: inst/WORDLIST [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+cyclomatic
+dcf
+de
+debian
+dplyr
+dragosmg
+eg
+envvar
+eval
+favour
```
Comment (MichaelChirico) [inline inst/WORDLIST:62]: same, we should be consistent

File: inst/WORDLIST [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+XPath
+XPaths
+YAML
+Yihui
+addin
+addins
+alllowercase
+backports
+backticked
+behaviour
```
Comment (MichaelChirico) [inline inst/WORDLIST:42]: another

File: inst/WORDLIST [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+SARIF
+SublimeLinter
+Tidyverse
+UpperCamelCase
+Wickham
+XPath
+XPaths
+YAML
+Yihui
+addin
```
Comment (MichaelChirico) [inline inst/WORDLIST:37]: is this an accepted spelling? I would have expected add-in

File: inst/WORDLIST [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+covr
+customizable
+customizations
+cyclomatic
+dcf
+de
+debian
+dplyr
+dragosmg
+eg
```
Comment (MichaelChirico) [inline inst/WORDLIST:59]: we should use `e.g.` consistently

File: inst/WORDLIST [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+repo
+rlang
+roxygen
+sandboxing
+spilt
+src
+stopifnot
+styler
+subdir
+syntastic
```
Comment (MichaelChirico) [inline inst/WORDLIST:102]: is syntastic a word?

Comment (IndrajeetPatil) [reply inst/WORDLIST:102]: It's name of a Vim plugin: https://github.com/vim-syntastic/syntastic

Comment (MichaelChirico) [reply inst/WORDLIST:102]: does `spelling` support comments in the `WORDLIST`? it could be good to annotate cases like this so we don't forget

Comment (IndrajeetPatil) [reply inst/WORDLIST:102]: IINM, it's a debian/control file, so, I don't think so.

File: inst/WORDLIST (hunk 3 above)
Comment (IndrajeetPatil) [reply inst/WORDLIST:37]: Yes: https://rstudio.github.io/rstudioaddins/

File: inst/WORDLIST [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+pre
+programmatically
+qmd
+readably
+regexes
+repo
+rlang
+roxygen
+sandboxing
+spilt
```
[the remark this answers is not in the document]
Comment (IndrajeetPatil) [reply inst/WORDLIST]: This is going to be a tough one for someone like me who comes from the Commonwealth. 😅

[the remark this answers is not in the document]
Comment (MichaelChirico) [reply inst/WORDLIST]: Just be happy we don't need to decide about metric units 😂

TBH I'm comfortable switching to `en-UK` too, but we should be consistent (do we have any function names assuming stateside spelling?).

File: inst/WORDLIST [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+io
+jwz
+knitr
+labelled
+lang
+languageserver
+linter
+linters
+lintr's
+localisation
```
Comment (MichaelChirico) [inline inst/WORDLIST:75]: one more 🇬🇧

File: inst/WORDLIST [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+lowerCamelCalse
+magrittr
+michaelchirico
+mis
+nd
+nodeset
+nolint
+parameterised
```
Comment (MichaelChirico) [inline inst/WORDLIST:83]: one more 🇬🇧

File: inst/WORDLIST (hunk 6 above)
[the remark this answers is not in the document]
Comment (IndrajeetPatil) [reply inst/WORDLIST]: Thankfully, no, we don't have any function names that assume US spellings.

**Review by MichaelChirico — APPROVED** [review]

**Pull request merged**